common-lib/src/main/scala/com/gu/mediaservice/lib/auth/ApiKey.scala

```diff
@@ -0,0 +1,32 @@
+package com.gu.mediaservice.lib.auth
+
+import com.gu.mediaservice.lib.argo.ArgoHelpers
+import play.api.mvc.Request
+
+sealed trait Tier
+case object Internal extends Tier
+case object External extends Tier
+object Tier {
+  def apply(value: String): Tier = value.toLowerCase match {
+    case "internal" => Internal
+    case "external" => External
+    case _ => External
+  }
+}
+
+case class ApiKey(value: String, tier: Tier)
```

Comment:
are we losing the name of the api key now?

Reply:
`value` is set to the first line in the S3 file. Bad naming... I can change it to `name` if it makes more sense.